actionview/lib/action_view/helpers/csrf_helper.rb

```diff
@@ -26,6 +26,12 @@ def csrf_meta_tags
             tag("meta", name: "csrf-token", content: form_authenticity_token)
           ].join("\n").html_safe
         end
+      rescue ActionDispatch::Request::Session::DisabledSessionError
+        # TODO: proper deprecation message
+        ActiveSupport::Deprecation.warn(<<-MSG.squish)
+          Calling `csrf_meta_tags` when session are disabled is deprecated. Either configure session or disable CSRF protection
+        MSG
+        nil
```

Comment:
I think this is a good example of why failing loudly would be preferable. Right now if you disable sessions but forget to disable CSRF protection, the tokens would still be generated but would always fail to validate.

Reply:
Another possibility is to not output any tags and log a debug warning.

This has echos of when we disabled autoload for the lib folder. Some people prefer hard failure and some people prefer soft warnings so I suspect it'll need to be configurable - hard failure for new applications and soft warnings for upgrading apps.

Reply:
> Another possibility is to not output any tags and log a debug warning.

Yes that's what I'm doing here as to preserve the current behavior (except it uses a deprecation warning rather than a regular one). I think that make sense when changing existing things.

> hard failure for new applications and soft warnings for upgrading apps.

Agreed.